app/controllers/devise_recognisable/sessions_controller.rb

```diff
@@ -42,6 +46,21 @@ def check_for_authentication_token
       self.resource = resource_class.find token['id']
 
       sign_in self.resource
+      DeviseRecognisable::RecognisableSession.create!(
+        recognisable: resource_class.find_by(email: self.resource.email),
+        sign_in_ip: request.location.ip,
+        sign_in_at: Time.now
+      )
     end
   end
+
+  # After the user has been signed in, we save the ip address in the
+  # DeviseRecognisable::RecognisableSession table.
+  def store_recognisable_details
```

Comment:
I'm not super familiar with circumstances where this after_action will or won't run. If you're going to keep this here (vs. just explicitly doing it on line 49 as above?), check (maybe inc. writing a test) that it doesn't get called if the sign-in fails.

Reply:
@jalada Tests like these https://github.com/pixielabs/devise-recognisable/pull/29/commits/b4063325737fd4070413b8cc0ec4d8f7d38d9db1?

Reply:
Yes